### PR TITLE
Minor fixes for SavedArticlesFetcher.swift

### PR DESCRIPTION
**Pull request by joewalsh**
Confirmed that this seems to be working (testing with the save buttons in the feed and the updated article view save button in the feature/mobile-html-ui branch)

### CODE REVIEW

File: Wikipedia/Code/SavedArticlesFetcher.swift [hunk 1]
```diff
@@ -76,7 +76,7 @@ private extension SavedArticlesFetcher {
     }
     
     @objc func articleWasUpdated(_ note: Notification) {
-        
+        update()
     }
     
     @objc func syncDidFinish(_ note: Notification) {
```
Comment (tonisevener) [inline Wikipedia/Code/SavedArticlesFetcher.swift]: @joewalsh I think I missed the update() call in syncDidFinish() as well (see LegacySavedArticlesFetcher.m)

Comment (joewalsh) [reply Wikipedia/Code/SavedArticlesFetcher.swift]: @tonisevener  good catch - fixed!

File: Wikipedia/Code/SavedArticlesFetcher.swift [hunk 2]
```diff
@@ -196,51 +199,61 @@ private extension SavedArticlesFetcher {
...
-                    switch groupResult {
-                    case .success:
+                    DispatchQueue.main.async {
+                        switch groupResult {
+                        case .success:
+                            print("🙈success in groupCompletion of \(articleKey)")
+                            self.didRemoveArticle(with: articleKey)
+                            self.updateFetchesInProcessCount()
+                        case .failure:
+                            print("🙈failure in groupCompletion of \(articleKey)")
```
Comment (tonisevener) [inline Wikipedia/Code/SavedArticlesFetcher.swift]: @joewalsh we might need a call here to self.updateFetchesInProcessCount()

Comment (tonisevener) [reply Wikipedia/Code/SavedArticlesFetcher.swift]: eh on second thought maybe not. If it fails to remove, isDownloaded will remain true thus the in process count won't change.

**Pull request merged**